papers/scipy/code/signal/bandpass_batch_example.py

```diff
@@ -0,0 +1,55 @@
+import numpy as np
+from scipy.signal import sosfreqz, butter, sosfilt
+import matplotlib.pyplot as plt
+from butter import butter_bandpass, butter_bandpass_filt
+from make_data import make_data
+
+
+# Sample rate and desired cutoff frequencies (in Hz).
+fs = 4800.0
+lowcut = 400.0
+highcut = 1200.0
+
+T = 0.06
+t, x = make_data(T, fs)
+
+sos = butter_bandpass(lowcut, highcut, fs, 12)
+
+batch_size = 72
+
+# Array of initial conditions for the SOS filter.
+z = np.zeros((sos.shape[0], 2))
+
+# Preallocate space for the filtered signal.
+y = np.empty_like(x)
+
+start = 0
+while start < len(x):
+    stop = min(start + batch_size, len(x))
+    y[start:stop], z = sosfilt(sos, x[start:stop], zi=z)
+    start = stop
+
+
+plt.figure(figsize=(4.0, 3.2))
+
+plt.plot(t, x, 'k', alpha=0.4, linewidth=1, label='Noisy signal')
+
+start = 0
+alpha = 0.5
+while start < len(x):
+    stop = min(start + batch_size, len(x))
+    if start == 0 or start == batch_size:
+        lbl = 'Filtered signal'
```

Comment:
nitpick: ``lbl`` is better named ``label``

Reply:
more importantly, the label ``Filtered signal`` now appears twice in the plot (light and dark blue), it would be better to only let it appear for dark blue (looks a bit odd to me now). The figure caption explains the rest.